backend/core/src/applications/applications.controller.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  Query,
+  UseGuards,
+  UseInterceptors,
+} from "@nestjs/common"
+import { ApplicationsListQueryParams } from "./applications.dto"
+import { ApplicationsService } from "./applications.service"
+import { Application } from "../entity/application.entity"
+import { JwtAuthGuard } from "../auth/jwt.guard"
+import { ApiBearerAuth } from "@nestjs/swagger"
+import { RolesGuard } from "../auth/roles.guard"
+import { Roles } from "../auth/roles.decorator"
+
+@Controller("applications")
+@UseGuards(JwtAuthGuard)
```

Comment:
> > @jaredcwhite their purpose is described in #130 .
> > > One seems to be for public access of data,
> > 
> > 
> > What do you mean by public? One must have an admin role to access `/applications`. This is created specifically for superusers querying over applications table.
> > > Can we simplify the API endpoint down to one and add logic there to allow or deny requests based on if auth is required?
> > 
> > 
> > Answer to this question depends on the answer to the previous one.
> 
> I guess what I'm driving at is I would expect a single API endpoint (`/applications`) to deal with the data model "applications". What that endpoint is allowed do (aka when querying or saving data) depends on the request auth. Let me know if we should hop on a chat to discuss further.
> 
> **Update:** sorry, I just noticed your link to #130. Maybe we can discuss with Ben then…I'm not convinced admin routes vs user-specific nested routes makes sense, it seems to make the codebase much more complex.

@jaredcwhite I find API easier to reason about when the endpoint semantics (instead of a query param) tell you that these applications are in a relation to a specific user. I don't have any strong arguments for that and an `/applications` with query params and request auth handling would probably do the work too. I just doubt that it would be less complex than what I proposed. Maybe we can engage more people into the discussion since we are probably going to create few similar endpoints in the future @software-project @bk3c @bencpeters 

BTW: `/applications` is optional and I created it as a convenience. Everything related to applications that we need at the moment is doable using `/users/id/applications`, correct? Superuser interface is not really specified at the moment and I created /applications just as a proposition of how such access to arbitrary application could work.

Reply:
@bencpeters As discussed elsewhere this must be JWT because roles are required on user object. Do you agree?